src/Factory/AbstractDsnFactory.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+/*
+ * This file is part of php-cache\adapter-bundle package.
+ *
+ * (c) 2015-2015 Aaron Scherer <aequasi@gmail.com>, Tobias Nyholm <tobias.nyholm@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Cache\AdapterBundle\Factory;
+
+use Cache\AdapterBundle\DSN;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+/**
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ */
+abstract class AbstractDsnFactory extends AbstractAdapterFactory
+{
+    /**
+     * @type DSN
+     */
+    private static $DSN;
+
+    protected static function getDsn()
+    {
+        return static::$DSN;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected static function configureOptionResolver(OptionsResolver $resolver)
+    {
+        $resolver->setDefaults(['dsn' => '']);
+        $resolver->setAllowedTypes('dsn', ['string']);
+    }
+
+    public static function validate(array $options, $adapterName)
+    {
+        parent::validate($options, $adapterName);
+
+        if (empty($options['dsn'])) {
+            return;
+        }
+
+        $dsn = new DSN($options['dsn']);
+        if (!$dsn->isValid()) {
+            throw new \InvalidArgumentException('Invalid DSN: '.$options['dsn']);
+        }
+
+        static::$DSN = $dsn;
```

Comment:
We do not need the DSN object in the factories except from the `AbstractDsnFactory`. We know the DSN is valid already so we can just use the string.